packages/edit-site/src/components/global-styles/style-variations-container.js

```diff
@@ -114,22 +119,110 @@ export default function StyleVariationsContainer() {
 				settings: {},
 				styles: {},
 			},
-			...( variations ?? [] ).map( ( variation ) => ( {
-				...variation,
-				settings: variation.settings ?? {},
-				styles: variation.styles ?? {},
-			} ) ),
+			...( variations ?? [] ).map( ( variation ) => {
+				const blockStyles = { ...variation?.styles?.blocks } || {};
+				if (
+					currentUserStyles?.styles?.blocks &&
+					preserveAdditionalCSS
+				) {
+					Object.keys( currentUserStyles.styles.blocks ).forEach(
+						( blockName ) => {
+							if (
+								currentUserStyles.styles.blocks[ blockName ].css
+							) {
+								blockStyles[ blockName ] = {
+									...( blockStyles[ blockName ]
+										? blockStyles[ blockName ]
+										: {} ),
+									css: `${
+										blockStyles[ blockName ]?.css || ''
+									} ${
+										currentUserStyles.styles.blocks[
+											blockName
+										].css
+									}`,
+								};
+							}
+						}
+					);
+				}
+
+				const styles = preserveAdditionalCSS
+					? {
+							...variation.styles,
+							...( currentUserStyles?.styles?.css ||
+							variation?.styles?.css
+								? {
+										css: `${
+											variation.styles?.css || ''
+										} ${ currentUserStyles.styles.css }`,
+								  }
+								: {} ),
+							blocks: {
+								...blockStyles,
+							},
+					  }
+					: variation.styles;
+
+				return {
+					...variation,
+					settings: variation.settings ?? {},
+					styles: styles ?? {},
+				};
+			} ),
 		];
-	}, [ variations ] );
+	}, [
+		variations,
+		currentUserStyles.styles.blocks,
+		currentUserStyles.styles.css,
+		preserveAdditionalCSS,
+	] );
+
```

Comment:
would be good to move the below to an event handler rather than useEffect, but need to refactor the update flows slightly to do this - will worry about that if we decide to go ahead with this refactor